Add __toString to a few methods

mallorydxw opened the pull request:
There's actually no method to get an IP range in the form `2001:db8::/64` out of the `IPv4Range` or `IPv6Range` classes. This would fix that.

- [x] Add `IPv*Address::FromBinary()` constuctors
- [x] Add `IPv*Block->__toString()` methods
- [x] Add `IPv*Range->__toString()` methods
- [x] Add extra tests for `IPv*Address::FromBinary()`
- [x] Add extra tests for `IPv*Range->__toString()`

mallorydxw committed:
BlockBase __toString

mallorydxw committed:
RangeBase __toString

mallorydxw committed:
Add FromBinary constructor

mallorydxw committed:
AddressBase & BlockBase should be abstract classes

RobjS commented:
@mallorydxw Should we add any documentation to the README about the new `FromBinary()` method (and particularly the class it expects the `$binary` argument to be)?

mallorydxw committed:
Add API documentation for FromBinary

mallorydxw committed:
Add generic notation to README for greater clarity

mallorydxw commented:
@RobjS I've addressed that oversight. I've also added some documentation showing what type of value is stored in the `Result` values that get returned.
